core/state/database.go

```diff
@@ -73,18 +74,24 @@ type Database interface {
 
 	Transitioned() bool
 
-	SetCurrentAccountHash(hash common.Hash)
+	SetCurrentSlotHash(hash common.Hash)
 
-	GetCurrentAccountHash() common.Hash
+	GetCurrentAccountAddress() common.Address
 
-	SetCurrentSlotHash(hash common.Hash)
+	SetCurrentAccountAddress(common.Address)
+
+	GetCurrentAccountHash() common.Hash
 
 	GetCurrentSlotHash() common.Hash
 
 	SetStorageProcessed(bool)
 
 	GetStorageProcessed() bool
 
+	GetCurrentPreimageOffset() int64
+
+	SetCurrentPreimageOffset(int64)
```

Comment:
TL;DR of changes:
- `SetCurrentAccountHash` was removed.
- `SetCurrentAccountAddress` was added, since we register the current address that we're walking to avoid "jumping back". No need to do that if we store it.
- `GetCurrentAccountHash` remains but is just a helper of doing `keccack(GetCurrentAccountAddress)`
- `(Set/Get)CurrentPreimageOffset` were added to register the current position in the preimage file, which only moves forward.